File: src/temporal_learn/process3/starter.py
import asyncio
import uuid

from temporalio.client import Client

from .models import ProductChangeSpec
from .workflow import ProductChangeWorkflow

TASK_QUEUE = "price-change-tq"


async def main():
    client = await Client.connect("localhost:7233")
    spec = ProductChangeSpec(
        request_id=f"REQ-{uuid.uuid7()}",
        product_id="P001",
        new_price=1490.0,
    )
    handle = await client.start_workflow(
        ProductChangeWorkflow.run,
        spec,
        id=f"price-change-{uuid.uuid7()}",
        task_queue=TASK_QUEUE,
    )
    print(f"WorkflowId: {handle.id}")


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/src/process3/starter.ts
import { Client, Connection } from '@temporalio/client';
import { randomUUID } from 'crypto';
import { TASK_QUEUE, TEMPORAL_URL, type ProductChangeSpec } from './models';
import { productChangeWorkflow } from './workflow';

async function main() {
  const connection = await Connection.connect({ address: TEMPORAL_URL });
  const client = new Client({ connection });
  const spec: ProductChangeSpec = {
    requestId: `REQ-${randomUUID()}`,
    productId: 'P001',
    newPrice: 1490,
  };

  const handle = await client.workflow.start(productChangeWorkflow, {
    args: [spec],
    taskQueue: TASK_QUEUE,
    workflowId: `price-change-ts-${randomUUID()}`,
  });

  console.log(`WorkflowId: ${handle.workflowId}`);
  await connection.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
